root/static/scripts/edit/URLCleanup.js

```diff
@@ -4497,154 +4591,317 @@ function testAll(tests, text) {
+      if (!check.result) {
+        return check;
+      }
+    }
+
+    const cleanup = this.cleanup;
+    const types = this.filterApplicableTypes(entityType);
+    // Check if given relationship type is applicable to entity
+    if (cleanup && types.length > 0) {
+      const validation = cleanup.validate
+        ? cleanup.validate(this.url, id)
+        : {result: true};
+      /*
+       * Check if given type is specified in `types`,
+       * either as a single type or in a combination.
+       */
+      const isRelationshipValid = types.some(
+        ids => typeof ids === 'string' ? ids === id : ids.includes(id),
+      );
+      validation.result = validation.result && isRelationshipValid;
+      return validation;
+    }
+    return {
+      result: RESTRICTED_LINK_TYPES.indexOf(id) === -1,
+      target: ERROR_TARGETS.RELATIONSHIP,
+    };
+  }
+
+  /*
+   * Validate relationship type combination.
+   * Should only be triggered after every single type
+   * has passed validation.
+   */
+  checkRelationships(
+    selectedTypes: Array<string>,
+    allowedTypes: Array<RelationshipTypeT> | false,
+  ): ValidationResult {
+    if (!allowedTypes) {
+      return {result: true};
     }
-    if (/^(https?:\/\/)?([^\/]+)\.bandcamp\.com\/?$/.test(url)) {
+    // Only a single type is selected
+    if (selectedTypes.length === 1) {
+      const type = selectedTypes[0];
+      const mustBeCombinedWith: Array<Array<string>> = [];
+      for (let i = 0; i < allowedTypes.length; ++i) {
+        const allowedType = allowedTypes[i];
+        if (type === allowedType) {
+          return {result: true};
+        }
+        /*
+         * Check if selected type in included in a combination
+         * If so, record other type(s) that're required
+         */
+        if (typeof allowedType === 'object' && allowedType.includes(type)) {
+          mustBeCombinedWith.push(
+            allowedType.filter(elem => elem !== type),
+          );
+        }
+      }
       return {
-        error: exp.l(
-          `This is a Bandcamp profile, not a page for a specific
-           release. Even if it shows this release right now,
-           that can change when the artist releases another.
-           Please find and add the appropriate release page
-           (“{album_url_pattern}” or “{single_url_pattern}”)
-           instead, and feel free to add this profile link
-           to the appropriate artist or label.`,
-          {
-            album_url_pattern: (
-              <span className="url-quote">{'/album'}</span>
-            ),
-            single_url_pattern: (
-              <span className="url-quote">{'/track'}</span>
-            ),
-          },
-        ),
         result: false,
-        target: ERROR_TARGETS.URL,
+        error: texp.l(
+          'This type must be used with {types}.',
+          {types: this.buildSentence(mustBeCombinedWith)},
+        ),
+        target: ERROR_TARGETS.RELATIONSHIP,
       };
```

Comment:
For example, when selecting the optional “lyrics” relationship type for a Mainly Norfolk URL, it returns “This type must be used with ‘other databases’.” just under the URL.

1. The actual cause of the error is not the selected relationship, it is the missing relationship. So the error target should not be `TARGET_ERRORS.RELATIONSHIP` and the error message should probably be something like “The relationship type ‘other databases’ is missing for this URL.”.
2. Maybe the error message could be displayed under the set of relationships, and even under the button “Add a new relationship” which can be used by the user to address this error. It seems it could be worth creating a new error target: `TARGET_ERRORS.RELATIONSHIP_SET`.

The 1. is a bug to be fixed, but the 2. is just a suggested improvement that can wait for further refactoring as [discussed on IRC](https://chatlogs.metabrainz.org/libera/metabrainz/msg/4850037/).

Reply:
Bug 1. is now fixed in commit <https://github.com/metabrainz/musicbrainz-server/pull/2184/commits/231b6c481e72e0e328901cc6183ec6698c6fb1fd>.

Reply:
Suggestion 2. was twofold: updating the error message will be part of MBS-11899, renaming `types` (and/or other properties of `CLEANUPS`?) will be decided at the same time as the larger renaming of `URLCleanup` and `ERROR_TARGETS` that was planned anyway.